import.py

```diff
@@ -667,10 +674,146 @@ def write_to_file(cont: str, filename: str) -> None:
+        temp_source = original_ctx + "\n\n" + original_source
+        write_to_file(temp_source, f"{dirname}/temp.c")
+        cpp_command = CPP + ["temp.c", "-D__sgi", "-D_LANGUAGE_C", "-DNON_MATCHING"]
+        no_macros_source = subprocess.check_output(
+            cpp_command + STUB_FN_MACROS,
+            cwd=dirname,
+            encoding="utf-8",
+            stderr=subprocess.DEVNULL,
+        )
+        source, context = prune_and_separate_context(no_macros_source, True, fn_name)
+
+        # Write out our base.c and ctx.h
+        write_to_file(context, f"{dirname}/ctx.h")
+        write_to_file('#include "ctx.h"\n' + source, f"{dirname}/base.c")
+
+    except Exception:
+        print(traceback.format_exc())
+        print("Exception occured while setting up base.c and ctx.h for the permuter")
+        sys.exit(1)
+
+    try:
+
+        compiler_id = response_json["compiler"].replace(".", "_")
+
+        if not os.path.exists("decompme_mappings.toml"):
+            print("decompme_mappings.toml not found")
+            print("See example_decompme_mappings.toml")
+            print("cp example_decompme_mappings.toml decompme_mappings.toml")
+            print("Then set specific properties for your compiler")
+            sys.exit(1)
+
+        with open("decompme_mappings.toml") as f:
+            all_settings = toml.load(f)
+
+        if compiler_id not in all_settings:
+            print(
+                f"Compiler {compiler_id} not found in decompme_compiler_mappings.toml"
+            )
+            print(
+                f"Add a new entry to this toml file, for this compiler: {compiler_id}"
+            )
+            print("See example_decompme_mappings.toml")
+            sys.exit(1)
+
+        compiler_settings = all_settings[compiler_id]
+        with open(compile_script, "w") as f:
+            f.write("#!/usr/bin/env bash\n")
+            f.write(f"cd {compiler_settings['PATH']}\n")
+
+            compile_line = "wine " if compiler_settings["USE_WINE"] else ""
+            compile_line += compiler_settings["EXE_NAME"] + " "
+            compile_line += response_json["compiler_flags"]
+            compile_line += " -c -o $3 $1"
+            f.write(compile_line)
+
+        os.chmod(compile_script, os.stat(compile_script).st_mode | stat.S_IEXEC)
+
+        compiler_type = compiler_settings.get("compiler_type", "base")
+
+        create_write_settings_toml(fn_name, compiler_type, settings_file)
+
+        print(
+            f"Success!! Prepared new folder, {dirname}, that is ready for the permuter script! Run with 'python3 permuter.py {dirname} [additional_flags]'"
+        )
+
+    except Exception:
+        print(traceback.format_exc())
+        print(
+            "Exception occured while creating a compile.sh for the permuter from decompme_mappings.toml and scratch settings"
+        )
+        sys.exit(1)
+
+
 def main() -> None:
+
+    if len(sys.argv) > 1 and sys.argv[1].startswith(DECOMPME_BASE):
```

Comment:
I'm assuming there's a way to do this that's more argparse-y instead of manually checking argv - that might be better here

Reply:
Not really afaik, we need to do different parsing depending on whether something that looks like a URL is passed